File: server/Scheduler.Interfaces/Models/SessionRequest.cs
using System;

namespace Scheduler.Interfaces.Models
{
    public class SessionRequest
    {
        public string Id { get; set; }
        public DateTime Start { get; set; }
        public DateTime End { get; set; }
        public string Title { get; set; }
        public string UserId { get; set; }
        public SessionRequestType SessionType { get; set; }
        public SessionRequestStatus Status { get; set; }
        public string SessionId { get; set; }
        public Session Session { get; set; }
    }
}

File: webapp/src/app/models/dto/SessionRequest.ts
export interface SessionRequest {
  id: string;
  title: string;
  start: Date;
  userId: string;
  sessionType: SessionRequestType;
  status: SessionRequestStatus;

  sessionId?: string;
  // TODO: Separate - non Dto related properties
  color: { primary: string; secondary: string };
}

export interface SessionRequestScheduled extends SessionRequest {
  end: Date;
  // TODO: Separate - non Dto related properties
  color: { primary: string; secondary: string };
  canJoin: boolean;
}

export interface SessionRequestInstant extends SessionRequest {
}

export enum SessionRequestStatus {
  Waiting,
  Scheduled,
  Started,
  Done,
}

export enum SessionRequestType {
  Instant,
  Scheduled,
}
